api/voice-react-native-sdk.api.md

```diff
@@ -845,7 +845,7 @@ export interface Voice {
     // @internal (undocumented)
     emit(voiceEvent: Voice.Event, listener: (...args: any[]) => void): boolean;
     // @internal (undocumented)
-    emit(voiceEvent: Voice.Event.AudioDevicesUpdated, audioDevices: AudioDevice[], selectedDevice: AudioDevice): boolean;
+    emit(voiceEvent: Voice.Event.AudioDevicesUpdated, audioDevices: AudioDevice[], selectedDevice: AudioDevice | null): boolean;

```

Comment:
Why not make `selectedDevice` optional instead?
>selectedDevice?: AudioDevice): boolean

Reply:
No strong reason for `null`, can be easily changed. What are the pros and cons of `undefined` vs. `null`?

Reply:
One thing I can imagine happening is default-valued function parameters. I don't think `null` gets overwritten, but `undefined` does, which I would say is a "pro" for `undefined` over `null`.

Reply:
It's more of an ease of use related really. When defining a function with params that can be null, it's better to make it optional that way you can call it using the following manner (needs verification)
```ts
foo(bar, baz);

// instead of
foo(bar, baz, null);
```

I understand in your case, it's an event but the mindset just applies the same.

Another reason is, you made it optional [here](https://github.com/twilio/twilio-voice-react-native/pull/131/files#diff-4682b6358cd5eb7f9f7d809200c4cd17864dcf78a845893ee093ff6fe5827e41R13). Just thinking more about consistency.

Reply:
I'm pretty settled on making it `undefined` (optional) instead of explicitly `null`, so I'll make the changes. For posterity though, I would like to point out that the `NativeAudioDevicesInfo` interface is what we expect from the native layer, and is separate from our public interfaces. In effect, the SDK will never emit or return that "native" interface.

Reply:
Gotcha. And sounds good on making it optional.